src/make.ts

```diff
@@ -350,10 +350,10 @@ export async function doBuildTarget(progress: vscode.Progress<{}>, target: strin
         await vscode.tasks.executeTask(myTask);
 
         const result: number = await(new Promise<number>(resolve => {
-            let disposable: vscode.Disposable = vscode.tasks.onDidEndTaskProcess(e => {
+            let disposable: vscode.Disposable = vscode.tasks.onDidEndTaskProcess((e: vscode.TaskProcessEndEvent) => {
                 if (e.execution.task.name === makefileBuildTaskName) {
                     disposable.dispose();
-                    resolve(e.exitCode);
+                    resolve(e.exitCode ?? ConfigureBuildReturnCodeTypes.cancelled);
```

Comment:
Did you actually try this? You cancelled the build via the bottom-right popup button, that caused the task to return undefined and so we now ensure we return cancelled?

Reply:
Followed up offline, changing to `other` due to the exitCode being 0 when the task is cancelled with the UI, and the handling of a cancelled task is done by the boolean `cancelBuild`, thus there isn't a need for having that here. I also tested and no telemetry gets duplicated either way. My expectation is that for all normal cases of a user trying to build, is that the `exitCode` will always be defined.